File: pset6/cash.py
from cs50 import get_float


def receberCentavos():
    while True:
        entrada = get_float('Digite seu troco: R$')

        if entrada > 0:
            entrada = entrada * 100
            return int(entrada)
        else:
            print(f'Somente números positivos.')

def contadorMoedas(valor, moeda):
    qtMoedas = 0
    while True:
        if valor >= moeda:
            valor = valor - moeda
            qtMoedas += 1
        else:
            return qtMoedas



entrada = receberCentavos()
print(f"Seus centavos: {entrada}")
moeda25 = contadorMoedas(entrada, 25)
entrada = entrada - (moeda25 * 25)
moeda10 = contadorMoedas(entrada, 10)
entrada = entrada - (moeda10 * 10)
moeda5 = contadorMoedas(entrada, 5)
entrada = entrada - (moeda5 * 5)
moeda1 = contadorMoedas(entrada, 1)
print(f"Quantidade de moedas recebidas: {moeda25+moeda10+moeda5+moeda1}.")

File: pset1/cash.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int qtMoedas(int valor, int moeda) {
    int moedas = 0;
    for(int valor_restante = valor; valor_restante >= moeda; valor_restante -= moeda) moedas++;
    return moedas;
}

int main(void){

  float troco;

  //Input positivo do usuário
  do {
    troco = get_float("Digite seu troco: ");
  } while (troco < 0);

  //Transformar o valor em reais para centavos
  int centavos = round((troco * 100));

  int moeda25 = qtMoedas(centavos, 25);
  centavos -= (moeda25 * 25);
  int moeda10 = qtMoedas(centavos, 10);
  centavos -= (moeda10 * 10);
  int moeda5 = qtMoedas(centavos, 5);
  centavos -= (moeda5 * 5);
  int moeda1 = qtMoedas(centavos, 1);
  centavos -= (moeda1 * 1);

  int qt_moedas = moeda25 + moeda10 + moeda5 + moeda1;

  printf("%i\n", qt_moedas);
}

